pkg/docker/mover.go

```diff
@@ -0,0 +1,50 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+)
+
+type DockerClient interface {
+	PullImage(ctx context.Context, image string) error
+	PushImage(ctx context.Context, image string, endpoint string) error
+	LoadFromFile(ctx context.Context, filepath string) error
+	SaveToFile(ctx context.Context, filepath string, images ...string) error
+}
+
+type (
+	ImageSourceLoader      func(ctx context.Context, client DockerClient, images ...string) error
```

Comment:
Should these just be modeled as an interface?

Reply:
Looking a little further: the `ImageMover` depends on a client simply to pass it forward. Why does the client need to be passed forward as opposed to created with a particular client?